src/controllers/authController.js

```diff
@@ -0,0 +1,72 @@
+const jwt=require('jsonwebtoken')
+const bcrypt=require('bcrypt')
+const User=require('../models/users.json')
+const fs=require('fs-extra')
+const path= require('path')
+const { v4: uuidv4 } = require('uuid');
+const { randomUUID } = require('crypto')
+
+var signup=(req,res)=>{
+    const userData={
+        id:randomUUID(),
+        email:req.body.email,
+        password:bcrypt.hashSync(req.body.password,8),
+        news_preferences:req.body.news_preferences
+    }
+
+    console.log(userData.id)
+
+    const write_path=path.join(__dirname,'..','models/users.json')
+
+    let current_users=JSON.parse(JSON.stringify(User))
```

Comment:
- Deep copying all the users in route level is not required. This can be moved to global level
- Use const instead of let for variables that are not reassigned.